libraries/ESP8266mDNS/ESP8266mDNS.cpp

```diff
@@ -539,15 +539,34 @@ void MDNSResponder::_parsePacket(){
     while (numAnswers--) {
       // Read name
       stringsRead = 0;
+      size_t last_bufferpos = 0;
       do {
         tmp8 = _conn_read8();
-        if (tmp8 & 0xC0) { // Compressed pointer (not supported)
-          tmp8 = _conn_read8();
-          break;
-        }
         if (tmp8 == 0x00) { // End of name
           break;
         }
+        if (tmp8 & 0xC0) { // Compressed pointer
+          uint16_t offset = ((((uint16_t)tmp8) & ~0xC0) << 8) | _conn_read8();
+          if (_conn->isValidOffset(offset)) {
+              last_bufferpos  = _conn->tell();
+#ifdef DEBUG_ESP_MDNS_RX
+              DEBUG_ESP_PORT.print("Compressed pointer, jumping from ");
+              DEBUG_ESP_PORT.print(last_bufferpos);
+              DEBUG_ESP_PORT.print(" to ");
+              DEBUG_ESP_PORT.println(offset);
+#endif
+              _conn->seek(offset);
+              tmp8 = _conn_read8();
```

Comment:
Are you reading tmp8 twice (here and in L568) ?

Reply:
Indeed, I forgot to remove the second one. Fixed.